src/test/java/stepdefs/SearchStepDefs.java

```diff
@@ -36,19 +35,40 @@ public void iAmOnTheGoogleUkHomepage() {
     driver.get("https://www.google.co.uk");
   }
   
-  @When("I enter a search term")
-  public void iEnterASearchTerm() {
+  @When("I enter a {string}")
+  public void iEnterA(String string) {
+    WebElement acceptAll = driver.findElement(SearchPage.ACCEPT_ALL);
+    acceptAll.sendKeys(Keys.ENTER);
     WebDriverWait wait = new WebDriverWait(driver, Duration.ofSeconds(15));
-    wait.until(ExpectedConditions.elementToBeClickable(By.xpath("//button[@id='L2AGLb2']"))).click();
-    driver.findElement(By.xpath("//input[@title='Search']")).sendKeys("BBC news");
-    driver.findElement(By.xpath("//input[@title='Search']")).sendKeys(Keys.ENTER);
+    WebElement searchButton = wait.until(ExpectedConditions.elementToBeClickable(SearchPage.SEARCH_BUTTON));
+    searchButton.click();
+  }
+
+  @When("entered a search term and clicked on I am Feeling Lucky")
+  public void enteredASearchTermAndClickedOnIAmFeelingLucky() {
+    WebElement acceptAll = driver.findElement(SearchPage.ACCEPT_ALL);
+    acceptAll.sendKeys(Keys.ENTER);
+    WebDriverWait wait = new WebDriverWait(driver, Duration.ofSeconds(15));
+    WebElement searchButton = wait.until(ExpectedConditions.elementToBeClickable(SearchPage.SEARCH_BUTTON));
+    searchButton.click();
+    searchButton.sendKeys("BBC");
+
+    Actions action = new Actions(driver);
+    WebElement luckyButton = driver.findElement(SearchPage.FEELING_LUCKY);
+    action.moveToElement(luckyButton).click().perform();
   }
 
   @Then("results relevant to the search term are returned")
   public void resultsRelevantToTheSearchTermAreReturned() {
-    List<WebElement> resultHeaders = driver.findElements(By.xpath("//a/h3"));
+    List<WebElement> resultHeaders = driver.findElements(SearchPage.RESULT_HEADERS);
```

Comment:
At the moment I think this is just returning an empty list because this locator is locating 0 elements, so therefore, the loop below isn't actually being executed, when you fix it to actually search with your search input in the earlier methods, you should see this correctly asserting. So yeah nothing to change here really except for you want to call the page object method, e.g. `new SearchPage().getResultHeaders();`

Reply:
Now its searching for each search term text